File: app/src/main/java/com/moh/mhealth/PatientDatabase.java
package com.moh.mhealth;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

@Database(entities = {Patient.class}, version = 1)
@TypeConverters({Converters.class})
public abstract class PatientDatabase extends RoomDatabase {
    public abstract PatientDao patientDao();

    private static PatientDatabase INSTANCE;

    public static PatientDatabase getDbInstance(Context context) {

        if (INSTANCE == null) {
            // Initialize database
            INSTANCE = Room.databaseBuilder(context,
                    PatientDatabase.class,
                    "patients").build();
        }

        return INSTANCE;
    }

}

File: app/src/main/java/com/moh/mhealth/PatientDatabase.kt
package com.moh.mhealth

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import kotlinx.coroutines.CoroutineScope

@Database(entities = [Patient::class], version = 1)
@TypeConverters(Converters::class)
abstract class PatientDatabase : RoomDatabase() {
    abstract fun patientDao(): PatientDao

    object DatabaseBuilder {
        private var INSTANCE: PatientDatabase? = null
        fun getDbInstance(context: Context, scope: CoroutineScope): PatientDatabase {
            return INSTANCE ?: synchronized(this) {
                INSTANCE = Room.databaseBuilder(context.applicationContext,
                    PatientDatabase::class.java,
                    "patients").build()
                INSTANCE!!
            }
        }
    }
}